### PR TITLE
Improved logging

### PR DESCRIPTION
**Pull request by npeterkamps**
**What**:

* fix: don't crash when jsdom's document is passed to prettyDOM.
* feat(logging): don't log DOM in Cypress, don't highlight in the browser.

**Why**:

* When passing in jsdom's `document`, `pretty-format` module crashes. (#44)
* When passing in Cypress `window.document` to `prettyDOM`, it crashes because `document.outerHTML.length` results in `Cannot read property length of undefined` error. (#44)
* As discussed, logging the DOM in Cypress isn't really useful, so the DOM logging has been removed in Cypress environment.

**How**:

* Use `document.documentElement` in `prettyDOM` to have the actual element with an `outerHTML` property.
* Refactored `debugDOM` function to determine the environment and how/whether to log the DOM element.

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Ready to be merged 
* [x] Added myself to contributors table

### CODE REVIEW

**Comment by aulisius** [comment]
Hey. Nice PR! Waiting on this currently. @kentcdodds yo, how soon can you merge this?

Thanks,
Faizaan.

File: src/queries.js [hunk 1]
```diff
@@ -3,7 +3,17 @@ import {getNodeText} from './get-node-text'
...
+  const inNode = (typeof module !== 'undefined' && module.exports)
+  const inBrowser = (typeof window !== 'undefined' && window.document)
+  const inCypress = (typeof window !== 'undefined' && window.Cypress)
+  if (inCypress) {
+    return ''
+  }
+  if (inBrowser && !inNode) {
+    return `\n\n${prettyDOM(htmlElement, limit, { highlight: false })}`
+  }
+  return `\n\n${prettyDOM(htmlElement, limit)}`
```
[the remark this answers is not in the document]
Comment (npeterkamps) [reply src/queries.js]: In Cypress, that would result in having two empty lines at the end of the error message. `debugDOM` seems to be a private function for query errors, so I thought this implementation would be preferable.

**Comment by npeterkamps** [comment]
> Looks like were missing some coverage. If it's really difficult to simulate a non-browser environment...

I'm wondering, in what kind of situation would `dom-testing-library` be running in a pure browser environment? We've handled Cypress' case. Codesandbox runs in a mixed environment and I think it should have the same results as Node for consistency ([as explain in this comment](https://github.com/kentcdodds/dom-testing-library/issues/44#issuecomment-392462886)). So maybe we don't need to support a browser-only environment and just have an exception for Cypress?

I'll look into the coverage issue, don't know why line 59 isn't being covered by the tests.

File: src/queries.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kentcdodds) [reply src/queries.js]: Updated my comment to avoid that issue

**Comment by npeterkamps** [comment]
Agreed. Changes pushed.

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Pull request merged**